tests/training/test_spacy.py

```diff
@@ -13,89 +13,98 @@
-    trainer = ArgillaTrainer(name=dataset_text_classification, model=MODEL, framework=FRAMEWORK)
-    trainer.update_config(max_steps=1)
-    output_dir = "tmp_spacy_train_textcat"
-    train_with_cleanup(trainer, output_dir)
-    record = trainer.predict("This is a text", as_argilla_records=True)
-    assert isinstance(record, rg.TextClassificationRecord)
-    assert record.multi_label is False
-    not_record = trainer.predict("This is a text", as_argilla_records=False)
-    assert isinstance(not_record.cats, dict)
-    train_with_cleanup(trainer, output_dir, train=False)
-    cleanup_spacy_config(trainer)
-
-
-def test_train_textcat_multi_label(dataset_text_classification_multi_label):
-    trainer = ArgillaTrainer(
-        name=dataset_text_classification_multi_label, model=MODEL, train_size=0.5, framework=FRAMEWORK
-    )
-    output_dir = "tmp_spacy_train_multi_label"
-    trainer.update_config(max_steps=1)
-    train_with_cleanup(trainer, output_dir)
-    record = trainer.predict("This is a text", as_argilla_records=True)
-    assert isinstance(record, rg.TextClassificationRecord)
-    assert record.multi_label is True
-    not_record = trainer.predict("This is a text", as_argilla_records=False)
-    assert isinstance(not_record.cats, dict)
-    train_with_cleanup(trainer, output_dir, train=False)
-    cleanup_spacy_config(trainer)
-
-
-def test_train_tokencat(dataset_token_classification):
-    trainer = ArgillaTrainer(name=dataset_token_classification, model=MODEL, framework=FRAMEWORK)
-    trainer.update_config(max_steps=1)
-    output_dir = "tmp_spacy_train_tokencat"
-    train_with_cleanup(trainer, output_dir)
-    record = trainer.predict("This is a text", as_argilla_records=True)
-    assert isinstance(record, rg.TokenClassificationRecord)
-    not_record = trainer.predict("This is a text", as_argilla_records=False)
-    assert not isinstance(not_record, rg.TokenClassificationRecord)
-    train_with_cleanup(trainer, output_dir, train=False)
-    cleanup_spacy_config(trainer)
-
-
-def test_init_with_gpu_id(dataset_text_classification):
-    trainer = ArgillaTrainer(name=dataset_text_classification, model=MODEL, framework=FRAMEWORK, gpu_id=0)
-    try:
-        import torch  # noqa
-
-        has_torch = True
-    except ImportError:
-        has_torch = False
-
-    try:
-        import tensorflow  # noqa
-
-        has_tensorflow = True
-    except ImportError:
-        has_tensorflow = False
-
-    import spacy
-
-    if not has_torch and not has_tensorflow:
-        assert trainer._trainer.gpu_id == -1
-    else:
-        assert trainer._trainer.use_gpu == spacy.prefer_gpu(0)
-
-
-def test_predict_wo_training(dataset_text_classification):
-    trainer = ArgillaTrainer(name=dataset_text_classification, framework=FRAMEWORK, model=MODEL)
-    trainer._trainer.predict("This is a text")
+from tests.training.helpers import cleanup_spacy_config, train_with_cleanup
+
+
+@pytest.mark.parametrize(
```

Comment:
You are not really passing the new variables in the tests right?

Reply:
Yes we are, as we included the args in the `test_` methods and then created the `TestSpaCyTrainer` class, so those values are passed to every function, and we're "looping" twice, once for `spacy` and another time for `spacy-transformers` 😄